### ISSUE TITLE
Interaction between an `enum` alias and `assert_never()` seems to fail

### ISSUE THREAD

**Issue by jhbuhrman** [python/mypy#21117]
**Bug Report**

If you have defined an `enum` with some values and an extra value which is an alias of an already defined one, the completeness-check using `assert_never(...)` does not work as (I) expected.

**To Reproduce**

```python
# file mypy_enum_alias.py
import enum

from typing import assert_never

class Version(enum.Enum):
    V1 = enum.auto()
    V2 = enum.auto()
    DEFAULT = V2

def print_version(version: Version) -> None:
    match version:
        case Version.V1:
            print("Version 1")
        case Version.V2:
            print("Version 2")
        case _:
            assert_never(version)

print(*(repr(v) for v in Version))
```

```shell
$ mypy mypy_enum_alias.py
```

**Expected Behavior**

```shell
Success: no issues found in 1 source file
```

**Actual Behavior**

```shell
mypy_enum_alias.py:18: error: Argument 1 to "assert_never" has incompatible type "Literal[Version.DEFAULT]"; expected "Never"  [arg-type]
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: `mypy 1.19.1 (compiled: yes)`
- Mypy command-line flags: (none)
- Mypy configuration options from `mypy.ini` (and other config files): 
  ```toml
  # pyproject.toml
  # ...
  [tool.mypy]
  plugins = [
      "pydantic.mypy"
  ]
  # ...
  ```
- Python version used: `Python 3.13.9`

If you run the script, Python bravely emits only the (two) non-aliased values:
```shell
$ python mypy_enum_alias.py
<Version.V1: 1> <Version.V2: 2>
```

If `mypy` considers all "names" of an `enum`, it renders the completeness check useless IMHO.

I could add the following code snippet just before `case _: ...` as a work-around, but that would just be pathetic:
```python
        case Version.DEFAULT:
            pass
```

But for the rest I'm quite happy with `mypy`. A great helper in early bug detection!

**Comment by jhbuhrman** [comment]
I'm not sure whether it relates to #8657, I guessed not...

**Comment by A5rocks** [comment]
Yeah I think this should work:
```
>>> import enum
>>> class Version(enum.Enum):
...     V1 = enum.auto()
...     V2 = enum.auto()
...     DEFAULT = V2
...
>>> Version.V2 == Version.DEFAULT
True
```

**Comment by jhbuhrman** [comment]
> Yeah I think this should work:
> 
> ```
> >>> import enum
> >>> class Version(enum.Enum):
> ...     V1 = enum.auto()
> ...     V2 = enum.auto()
> ...     DEFAULT = V2
> ...
> >>> Version.V2 == Version.DEFAULT
> True
> ```

Yes, it does!

Even more so, these are aliases to the very same object (identy equality vs. value equality):
```
>>> import enum
>>> class Version(enum.Enum):
...     V1 = enum.auto()
...     V2 = enum.auto()
...     DEFAULT = V2
...
>>> Version.V2 is Version.DEFAULT
True
```
